packages/atlas/src/themesource/atlas_core/web/layouts/_layout-atlas-responsive.scss

```diff
@@ -70,34 +73,38 @@
             }
         }
 
-        .mx-scrollcontainer-slide {
-            &:not(.mx-scrollcontainer-open) > .region-sidebar {
-                overflow: hidden;
-            }
+        @if (not $use-modern-client) {
+            // this is hacking dojo scroll container
+            // in react it is not needed
+            .mx-scrollcontainer-slide {
```

Comment:
Do we still need this comments? What part is dojo? What we hacking?

Reply:
I removed the comment.

Some parts of this file apply invasive styling to the dojo-based scroll container by knowing implementation details and abusing those. For example, forcing region width with !important and overriding what scroll container implementation itself puts in it's inline styling. I can't call anything but hacking :) And it was there for years to achieve functionality that didn't exist on the scroll container level itself (partially-visible sidebar).

And the react-based scroll container is working in a different way. This behavior that atlas is trying to implement by hacking dojo is now part of the functionality of the react-based scroll container, so it works ootb without hacking.

Reply:
Thx for explanation.